File: routeMap/app/src/main/java/ru/android/zheka/coreUI/IActivity.java
package ru.android.zheka.coreUI;

import android.app.Activity;
import android.content.Context;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;

public interface IActivity {
    Activity getActivity();

    void showError(Throwable throwable);

    Context getContext();

    void switchToFragment(int fragmentId, @NonNull Fragment fragment);

    void removeFragment(@NonNull Fragment fragment);

    FragmentManager getManager();
}

File: routeMap/app/src/main/java/ru/android/zheka/coreUI/IActivity.kt
package ru.android.zheka.coreUI

import android.app.Activity
import android.content.Context
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager

interface IActivity {
    open fun context():Context
    open val activity: Activity
    open fun showError(throwable: Throwable)
    open fun switchToFragment(fragmentId: Int, fragment: Fragment)
    open fun removeFragment(fragment: Fragment)
    open val manager: FragmentManager?
}